lib/demon/sidekiq.rb

```diff
@@ -3,6 +3,49 @@
 require "demon/base"
 
 class Demon::Sidekiq < Demon::Base
+  RANDOM_HEX = SecureRandom.hex
+
+  def self.heartbeat_queues_list_key
+    @@heartbeat_queues_list_key ||= "#{RANDOM_HEX}_heartbeat_queues_list"
+  end
+
+  def self.queues_last_heartbeat_hash_key
+    @@queues_last_heartbeat_hash_key ||= "#{RANDOM_HEX}_queues_last_heartbeat_hash"
+  end
+
+  def self.set_queue_heartbeat(name)
+    $redis.hset(queues_last_heartbeat_hash_key, name, Time.new.to_i.to_s)
+  end
+
+  def self.get_queue_last_heartbeat(name)
+    $redis.hget(queues_last_heartbeat_hash_key, name)&.to_i || 0
```

Comment:
You don't need the `&.` or the `|| 0` when using `to_i` since `nil.to_i == 0` 😉 

```suggestion
    $redis.hget(queues_last_heartbeat_hash_key, name).to_i
```